src/SWP/Bundle/TemplateEngineBundle/Command/ThemeGenerateCommand.php

```diff
@@ -0,0 +1,288 @@
+            $this->writeConfigFile($input, $output, $themeDir, $themeName);
+            $this->updateTenantReferenceToTheme($tenant, $themeName);
+            $output->writeln('Theme '.$themeName.' has been generated successfully!');
+        } catch (\Exception $e) {
+            $output->writeln('Theme '.$themeName.' could not be generated!');
+            $output->writeln($e->getMessage());
+        }
+    }
+
+    /**
+     * Gets the tenant based on the input - prompts a user to choose an existing tenant of the given organisation, or to create a new one.
+     *
+     * @param InputInterface  $input  An InputInterface instance
+     * @param OutputInterface $output An OutputInterface instance
+     *
+     * @return mixed|null
+     *
+     * @throws \Exception
+     */
+    protected function getTenant(InputInterface $input, OutputInterface $output)
+    {
+        $organizationRepository = $this->getContainer()->get('swp.repository.organization');
+        $organizationName = $input->getArgument('organizationName');
+        $organization = $organizationRepository->findOneByName($organizationName);
+        if (null === $organization) {
+            throw new OrganizationNotFoundException($organizationName);
+        }
+
+        $helper = $this->getHelper('question');
+        $question = new ConfirmationQuestion('Create new tenant?', false, '/^(y|j)/i');
+        if (!$helper->ask($input, $output, $question)) {
+            $tenants = $organization->getTenants()->toArray();
+            $numTenants = count($tenants);
+            if (!$numTenants) {
+                throw new \Exception('Organization has no tenants');
+            }
+
+            $tenant = reset($tenants);
+            if ($numTenants > 1) {
+                $tenantNames = array_map(function ($tenant) {
+                    return $tenant->getName();
+                }, $tenants);
+
+                $tenants = array_combine($tenantNames, $tenants);
+
+                $question = new ChoiceQuestion(
+                    'Please select a tenant',
+                    $tenantNames,
+                    $tenantNames[0]
+                );
+                $question->setErrorMessage('Name %s is not an option');
+
+                $tenantName = $helper->ask($input, $output, $question);
+                $tenant = $tenants[$tenantName];
+            }
+
+            return $tenant;
+        } else {
+            $output->writeln('Creation of tenant here still to be implemented - you can create a tenant using the swp:tenant:create command');
+        }
+    }
+
+    /**
+     * Creates folders and empty files of theme.
+     *
+     * @param $tenantCode
+     * @param $themeName
+     */
+    protected function createSkeleton(Filesystem $fileSystem, $tenantCode, $themeName)
+    {
+        $configFilename = $this->getContainer()->getParameter('sylius.theme.configuration.filename');
```

Comment:
By getting this parameter here you make a reference to `CoreBundle` that means that you will need to require CoreBundle into TemplateEngineBundle just to get two parameters. I would move that command to `CoreBundle`.